sky/utils/controller_utils.py

```diff
@@ -895,6 +895,7 @@ def _sub_path_join(sub_path: Optional[str], path: str) -> str:
         # file_mount_remote_tmp_dir will only exist when there are files in
         # the src for copy mounts.
         storage_obj = task.storage_mounts[file_mount_remote_tmp_dir]
+        # TODO(zhwu): this fails due to IndexError: list index out of range
```

Comment:
btw i think all of step 5 can just be deleted. `task.sync_storage_mounts` already writes the updated file mounts